app/controllers/simulations_controller.rb

```diff
@@ -194,6 +196,14 @@ def mark_as_complete
             unless sm_record.computational_resources.blank?
               @simulation_run.computational_resources = sm_record.computational_resources
             end
+
+            sm_record.finished_simulations ||= 0
+            sm_record.finished_simulations += 1
+            sm_record.save
+          end
+
+          if params.include? :execution_statistics
```

Comment:
params[:execution_statistics] nie jest walidowane

To może być tylko:
- String, który jest JSON-em
- obiekt Hash

Walidacja może wyglądać tak:

```
execution_statistics = Utils.parse_json_if_string(params[:execution_statistics])
raise ValidationError.new(...) unless execution_statistics.nil? or execution_statistics.kind_of?(Hash)
```

Dodatkowa uwaga do stylu kodowania: https://github.com/bbatsov/ruby-style-guide#no-dsl-parens - bo widzę, że chyba @mwrona często omija nawiasy